scss/_variables.scss

```diff
@@ -486,14 +488,16 @@ $input-plaintext-color:                 $body-color !default;
 
 $input-height-border:                   $input-border-width * 2 !default;
 
-$input-height-inner:                    ($input-btn-font-size * $input-btn-line-height) + ($input-btn-padding-y * 2) !default;
-$input-height:                          calc(#{$input-height-inner} + #{$input-height-border}) !default;
+$input-height-inner:                    calc(#{$input-btn-line-height * 1em} + #{$input-btn-padding-y * 2}) !default;
+$input-height-inner-half:               calc(#{$input-btn-line-height * .5em} + #{$input-btn-padding-y}) !default;
+$input-height-inner-quarter:            calc(#{$input-btn-line-height * .25em} + #{$input-btn-padding-y / 2}) !default;
+$input-height:                          calc(#{$input-btn-line-height * 1em} + #{$input-btn-padding-y * 2} + #{$input-height-border}) !default;
 
-$input-height-inner-sm:                 ($input-btn-font-size-sm * $input-btn-line-height-sm) + ($input-btn-padding-y-sm * 2) !default;
-$input-height-sm:                       calc(#{$input-height-inner-sm} + #{$input-height-border}) !default;
+$input-height-inner-sm:                 calc(#{$input-btn-line-height-sm * 1em} + #{$input-btn-padding-y-sm * 2}) !default; // This variable is not used by Bootstrap but kept for backwards compatibility, TODO: remove in v5
```

Comment:
@mdo, I've kept `$input-height-inner-sm` and `$input-height-inner-lg` for backwards compatibility but maybe we can delete them? It's a variable used to calculate another variable, not sure if it's really a breaking change if we remove this?